### PR TITLE
Draft PR to share BUILD.bazel

### PR DESCRIPTION
**Pull request by jeffng-or**
Trying to pass MACROS to floorplan stage, such that it uses the BLOCKS_grid_strategy.tcl in PDN GEN.

### CODE REVIEW

**Review by oharboe — CHANGES_REQUESTED** [review]
If this is not working, it must be fixed in bazel-orfs

**Comment by jeffng-or** [comment]
After pouring over bazel-orfs, the megaboom BUILD.bazel and gcd BUILD.bazel, I realized that the MACROS path won't work. I also finally saw that you forced the PDN_TCL in the mega-array example, so that's what I'm going to try next.

I'm also starting to document both the megaboom BUILD.bazel and parts of bazel-orfs' openroad.bzl to help explain things as I dig into it. We can chat about it when we meet up next week.

File: BUILD.bazel [hunk 1]
```diff
@@ -257,7 +258,7 @@ orfs_flow(
             "GPL_ROUTABILITY_DRIVEN": "1",
             "GPL_TIMING_DRIVEN": "0",
         },
-        "route": SKIP_REPORT_METRICS,
+	"route": SKIP_REPORT_METRICS | { "GENERATE_ARTIFACTS_ON_FAILURE": "1" },
```
Comment (oharboe) [inline BUILD.bazel:261]: Practically, this won't do anything until we have a new version in ORFS that breaks out grt in a separate stage and it should be set by bazel-orfs.

Running a build now. I'll merge this sans this last line, if it works, as a new baseline.

Comment (oharboe) [reply BUILD.bazel:261]: Hmmm... actually, these two PDN strategies go together as a pair. One for submacros and another for the top level, so further changes are needed...

```
$ grep -r PDN_TCL ../OpenROAD-flow-scripts/flow/designs/asap7/mock-array/
../OpenROAD-flow-scripts/flow/designs/asap7/mock-array/config.mk:  export PDN_TCL               = $(PLATFORM_DIR)/openRoad/pdn/BLOCKS_grid_strategy.tcl
../OpenROAD-flow-scripts/flow/designs/asap7/mock-array/Element/config.mk:export PDN_TCL                = $(FLOW_HOME)/platforms/asap7/openRoad/pdn/BLOCK_grid_strategy.tcl

```

**Review by oharboe — CHANGES_REQUESTED** [review]
Set the submacro and top level PDN_TCL respectively for all macros in megaboom:

```
$ grep -r PDN_TCL mock-array/
../OpenROAD-flow-scripts/flow/designs/asap7/mock-array/config.mk:  export PDN_TCL=$(PLATFORM_DIR)/openRoad/pdn/BLOCKS_grid_strategy.tcl
mock-array/Element/config.mk:export PDN_TCL=$(FLOW_HOME)/platforms/asap7/openRoad/pdn/BLOCK_grid_strategy.tcl
```

**Pull request closed without merging**